addons/website/static/src/js/content/snippets.animation.js

```diff
@@ -1155,6 +1155,280 @@ registry.anchorSlide = publicWidget.Widget.extend({
+    selector: '.s_countdown',
+    disabledInEditableMode: false,
+
+    /**
+     * @override
+     */
+    start: function () {
+        var $el = this.$el;
+        this.endAction = $el.attr('data-end-action');
+        this.endTime = parseInt($el.attr('data-end-time'));
+        this.size = $el.attr('data-size');
+        this.layout = $el.attr('data-layout');
+        this.dataDisplay = this.$el.attr('data-display');
+
+        this._initTimeDiff();
+        this._drawCountdown();
+
+        // Needed if edit mode entered after countdown end
+        this.$el.find('.s_countdown_end_redirect_message').addClass('d-none');
+
+        // Show (non editable) msg when user lands on already finished countdown
+        if (this.isFinished && this.endAction === 'redirect' && !this.editableMode) {
+            this.$el.find('.s_countdown_end_redirect_message').removeClass('d-none');
+            return this._super.apply(this, arguments);
+        }
+
+        // Show end message event if not finished so it can be edited.
+        this.$el.find('.s_countdown_end_message').toggleClass('d-none', !(this.editableMode && this.endAction === 'message'));
+
+        this.setInterval = setInterval(this._manageCountdown.bind(this), 1000);
+        return this._super.apply(this, arguments);
+    },
+    /**
+     * @override
+     */
+    destroy: function () {
+        clearInterval(this.setInterval);
+        this._super(...arguments);
+    },
+
+    //--------------------------------------------------------------------------
+    // Private
+    //--------------------------------------------------------------------------
+
+    /**
+     * Draws the whole countdown, including one countdown for each time unit.
+     *
+     * @private
+     */
+    _drawCountdown: function () {
+        this._updateTimeDiff();
+        _.each(this.diff, (val, index) => {
+            var canvas = val['canvas'];
+            var ctx = canvas.getContext("2d");
+            ctx.canvas.width = this.size;
+            ctx.canvas.height = this.size;
+            this._clearCanvas(ctx);
+
+            $(canvas).toggleClass('d-none', !val['visible']);
+            if (!val['visible']) {
+                return;
+            }
+
+            if (this.layout === 'dark') {
+                this._drawBgCircle(ctx, true);
+            }
+            if (this.layout === 'thin-filled') {
+                this._drawBgCircle(ctx);
+            }
+            this._drawText(canvas, val['nb'], index);
+            if (this.layout === 'light') {
+                this._drawOuterCircleBg(ctx);
+            }
+            this._drawOuterCicle(ctx, val['nb'], val['total'], this.layout === 'thin' || this.layout === 'thin-filled');
+        });
+    },
+    /**
+     * Initializes the `diff` object. It will contains for every time unit the
+     * related canvas, its total step and its visibility.
+     *
+     * @private
+     */
+    _initTimeDiff: function () {
```

Comment:
What is better with it? I am in the start, then I must go elsewhere to see a simple object being instantiated... but if you want it, as you want :p

Reply:
Obviously the first time you read the code it adds an extra step since you have to check what the method does, but once you know `_initTimeDiff()` inits the `timeDiff` object which is the core of the countdown widget, I feel like it cleans the `start` method by encapsulating those 30 lines of code.

But I changed the code and put them back in the start, it is a detail anyway.